### PR TITLE
Cortx 33332

### PR DESCRIPTION
**Pull request by sonalk0209**
# Problem Statement
- test-40389 added

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/control_pod/test_control_pod_restart.py [hunk 1]
```diff
@@ -865,29 +857,35 @@ def test_ctrl_pod_shutdown_loop(self):
...
+        Verify IAM user and bucket operations while N-1 control pods are restarted
         """
-        LOGGER.info("STARTED: Verify IAM user and bucket operations while control pod is failing "
-                    "over")
+        LOGGER.info("STARTED: Verify IAM user and bucket operations while N-1 control pods "
+                    "are restarted")
+        LOGGER.info("Prereq: Scale replicas for control pod to %s", self.repl_num)
+        pod_name = self.node_master_list[0].get_all_pods(
+            pod_prefix=const.CONTROL_POD_NAME_PREFIX)[0]
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
```
Comment (ankita-93) [inline tests/ha/control_pod/test_control_pod_restart.py:869]: pod replicas of only one control pod ?

File: tests/ha/control_pod/test_control_pod_restart.py [hunk 2]
```diff
@@ -910,33 +907,31 @@ def test_iam_bkt_cruds_during_ctrl_pod_failover(self):
...
+        LOGGER.info("Step 3: Restart %s control pods while creation/deletion of IAM users and "
+                    "buckets is running in background.", self.repl_num - 1)
+        for count in range(HA_CFG["common_params"]["short_loop"]):
+            LOGGER.info("Restarting %s control pods for loop : %s", self.repl_num - 1, count + 1)
+            ctrl_pod_list = self.system_random.sample(pod_list, self.repl_num - 1)
+            for pod in ctrl_pod_list:
+                resp = self.node_master_list[0].delete_pod(pod)
+                assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 3.1: Check cluster status")
+            delay = HA_CFG["common_params"]["60sec_delay"]
```
Comment (ankita-93) [inline tests/ha/control_pod/test_control_pod_restart.py:919]: We can directly send this to timeout if delay is not used anywhere else.

File: tests/ha/control_pod/test_control_pod_restart.py (hunk 1 above)
Comment (sonalk0209) [reply tests/ha/control_pod/test_control_pod_restart.py:869]: Actually by default we would have only one control pod and then we scale it for multiple replicas, but maybe yes, we may need to add something in teardown to come back to one if more than one

File: tests/ha/control_pod/test_control_pod_restart.py (hunk 2 above)
Comment (sonalk0209) [reply tests/ha/control_pod/test_control_pod_restart.py:919]: The line was going beyond limit and don't wanted to break so added a var :)

File: tests/ha/control_pod/test_control_pod_restart.py [hunk 3]
```diff
@@ -910,33 +907,31 @@ def test_iam_bkt_cruds_during_ctrl_pod_failover(self):
...
+            resp = self.ha_obj.poll_cluster_status(self.node_master_list[0], timeout=delay)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 3.1: Cluster status is online")
+        LOGGER.info("%s pod are restarted successfully in loop of %s", self.repl_num - 1,
+                    HA_CFG["common_params"]["short_loop"])
         event.clear()
-
-        self.restore_pod = self.deploy = True
-        LOGGER.info("Step 4: Verify if IAM users %s are persistent across control pod failover",
+        LOGGER.info("Step 4: Verify if IAM users %s are persistent across control pods restart",
```
Comment (apurwamohite) [inline tests/ha/control_pod/test_control_pod_restart.py:926]: Should we also check that all background operations are successful as 1 control pod will always be alive?

Comment (sonalk0209) [reply tests/ha/control_pod/test_control_pod_restart.py:926]: After looking at this, I guess originally we had written this step even before joining other threads. Should we then shift this afterwards after all background activity is done? So all validation will be in one go?

**Comment by sonalk0209** [comment]
[pr_1846.log](https://github.com/Seagate/cortx-test/files/9444374/pr_1846.log)
collection log

**Review by ankita-93 — APPROVED** [review]

**Review by apurwamohite — APPROVED** [review]

**Review by apurwamohite — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**